spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -109,6 +144,90 @@ public void setTablePrefix(String tablePrefix) {
 		this.tablePrefix = tablePrefix;
 	}
 
+	@Override
+	public TaskExecution getTaskExecution(String executionId) {
+		try {
+			TaskExecution taskExecution = jdbcTemplate.queryForObject(getQuery(GET_EXECUTION_BY_ID),
+					new TaskExecutionRowMapper(), executionId);
+			taskExecution.setParameters(getTaskParameters(executionId));
+			return taskExecution;
+		}
+		catch (EmptyResultDataAccessException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public long getTaskExecutionCount(String taskName) {
+		try {
+		return jdbcTemplate.queryForObject(
+				getQuery(TASK_EXECUTION_COUNT), new Object[] { taskName }, Long.class);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return 0;
+		}
+
+	}
+
+	@Override
+	public Set<TaskExecution> findRunningTaskExecutions(String taskName) {
+		final Set<TaskExecution> result = new HashSet<TaskExecution>();
+		RowCallbackHandler handler = new RowCallbackHandler() {
+			@Override
+			public void processRow(ResultSet rs) throws SQLException {
+				TaskExecutionRowMapper mapper = new TaskExecutionRowMapper();
+				result.add(mapper.mapRow(rs, 0));
+			}
+		};
+		jdbcTemplate.query(getQuery(FIND_RUNNING_TASK_EXECUTIONS),
+				new Object[] { taskName }, handler);
+
+		return result;
+
+	}
+
+	@Override
+	public List<TaskExecution> getTaskExecutionsByName(String taskName, final int start, final int count) {
+		ResultSetExtractor<List<TaskExecution>> extractor =
+				new ResultSetExtractor<List<TaskExecution>>() {
+
+			private List<TaskExecution> list = new ArrayList<TaskExecution>();
+
+			@Override
+			public List<TaskExecution> extractData(ResultSet rs) throws SQLException,
+					DataAccessException {
+				int rowNum = 0;
+				while (rowNum < start && rs.next()) {
+					rowNum++;
+				}
+				while (rowNum < start + count && rs.next()) {
+					RowMapper<TaskExecution> rowMapper = new TaskExecutionRowMapper();
+					list.add(rowMapper.mapRow(rs, rowNum));
+					rowNum++;
+				}
+				return list;
+			}
+
+		};
+
+		List<TaskExecution> result = jdbcTemplate.query(getQuery(FIND_TASK_EXECUTIONS),
+				new Object[] { taskName }, extractor);
+
+		return result;
+	}
+
+	@Override
+	public List<String> getTaskNames() {
+		return jdbcTemplate.query(getQuery(FIND_TASK_NAMES),
+				new RowMapper<String>() {
+					@Override
+					public String mapRow(ResultSet rs, int rowNum)
+							throws SQLException {
+						return rs.getString(1);
+					}
+				});
```

Comment:
I would try `queryForList` passing in `String.class` as the type -- see http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/jdbc/core/JdbcTemplate.html#queryForList-java.lang.String-java.lang.Class-